### PR TITLE
Make maphammer tests easier to use

### PR DESCRIPTION
**Pull request by daviddrysdale**
Best reviewed commit-by-commit

### CODE REVIEW

File: testonly/hammer/hammer.go [hunk 1]
```diff
@@ -178,6 +178,17 @@ func (c MapConfig) String() string {
...
 
+	if cfg.MapID == 0 {
+		// No mapID provided, so create an ephemeral tree to test against.
+		var err error
+		cfg.MapID, err = makeNewMap(ctx, cfg.Admin, cfg.Client)
+		if err != nil {
+			return fmt.Errorf("failed to create ephemeral tree: %v", err)
+		}
+		glog.Infof("testing against ephemeral tree %d", cfg.MapID)
+		defer destroyMap(ctx, cfg.Admin, cfg.MapID)
```
Comment (golangci) [inline testonly/hammer/hammer.go]: Error return value of `destroyMap` is not checked

**Comment by daviddrysdale** [comment]
(Might be worth dropping the commit with in-process run of maphammer -- when I play with this on OSX I have problems with FD limits as well as connection counts...)

File: scripts/connlimit.sh [hunk 2]
```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
```
Comment (gdbelvin) [inline scripts/connlimit.sh]: would `scripts/mysqlconnlimit.sh` be a better name?

File: testonly/hammer/hammer.go (hunk 1 above)
Comment (gdbelvin) [reply testonly/hammer/hammer.go]: check the return value?

File: testonly/hammer/maphammer/main.go [hunk 3]
```diff
@@ -43,7 +43,7 @@ import (
 )
 
 var (
-	mapIDs          = flag.String("map_ids", "", "Comma-separated list of map IDs to test")
+	mapIDs          = flag.String("map_ids", "", "Comma-separated list of map IDs to test; ephemeral tree used if empty")
```
Comment (gdbelvin) [inline testonly/hammer/maphammer/main.go]: :-)

File: scripts/connlimit.sh (hunk 2 above)
Comment (daviddrysdale) [reply scripts/connlimit.sh]: Yep, done.

File: testonly/hammer/hammer.go (hunk 1 above)
Comment (daviddrysdale) [reply testonly/hammer/hammer.go]: Done.

**Pull request merged**